src/components/MultiSelect/MultiSelect.tsx

```diff
@@ -244,6 +251,39 @@ function MultiSelect<T>({
         setSelectedItems(externalSelectedItems);
     }, [externalSelectedItems?.map((item) => itemId(item)).join("|")]);
 
+    React.useEffect(() => {
+        if (!maxHeight) return;
+
+        const maxHeightToProcess = typeof maxHeight === "number" ? maxHeight : 100;
+
+        const calculateMaxHeight = () => {
+            if (inputRef.current) {
+                const viewportHeight = window.innerHeight;
+
+                // Get the distance from the top of the page to the bottom of the input
+                const dropdownHeight =
+                    inputRef.current.getBoundingClientRect().top + inputRef.current.getBoundingClientRect().bottom;
+
+                const availableSpace = viewportHeight - dropdownHeight;
```

Comment:
The calculation may be not correct because with `maxHeight=true` I see a set --multisuggest-max-height of "100vh" what cannot be correct, it must always be smaller.

Calculation could probably also done in CSS like `calc(#maxHeightOr100#vh - MultiSuggestFieldTargetHeightInPixels)` (pseudo code!, replace vars by correct values). Another option could be to do set two custom properties and then do the math in the stylesheet (e.g. `max-height: calc(var(--multisuggest-max-height, 45vh) - var(--eccgui-multiselectfield-currenttargetheight, 0px)`). Please test this.

Reply:
I tested this, neither of options worked for me. Maybe if you have some time you could also give it a try.
I improved the initial solution though, so now it seems to be working properly.